### DISCUSSION TITLE
[Projects Beta] Support "Multi select" field

### DISCUSSION THREAD

**Discussion by creasty** [community/community#6580] [Projects and Issues] [answered]
> [!IMPORTANT]
> ### Read the latest update from a GitHub staff:<br>https://github.com/orgs/community/discussions/6580#discussioncomment-4887894

<img width="347" alt="Screen Shot 2021-10-21 at 18 31 42" src="https://user-images.githubusercontent.com/1695538/138251076-22db0fbd-805f-4215-b3b4-30ad13ed4b42.png">

In my company, for instance, we often create issues that span different areas of development (i.e. design, frontend, backend, etc), and we would like to manage this attribute as a multi-select field.
Ideally, those issues can be divided into smaller subordinate issues for each area, but we tend to do so as we go.

This could also benefit the scenario where users want to manage issues that concern multiple teams.

**Comment by edmundmaruhn** [comment]
I'm currently working on a larger application and we divide it into broader components. To reflect these components we use a single-select, but several times issues arise which are related to multiple components at the same time.

A multi-select would better reflect this instead of insanely adding multiple single-selects or circumvent the absence of a multi-select by messing up issue labels which would have to be duplicated into multiple repositories.

**Reply by Julian-Dumitrascu** [reply]
We don't need to explain this to developer 1 when developer 2 is offering this feature. Developer 1 should just include it themselves.

**Reply by Pravinhajare** [reply]
qk

**Reply by Pravinhajare** [reply]
> I'm currently working on a larger application and we divide it into broader components. To reflect these components we use a single-select, but several times issues arise which are related to multiple components at the same time.
> 
> A multi-select would better reflect this instead of insanely adding multiple single-selects or circumvent the absence of a multi-select by messing up issue labels which would have to be duplicated into multiple repositories.

kk

**Reply by Pravinhajare** [reply]
> qk

ok

**Reply by Pravinhajare** [reply]
> We don't need to explain this to developer 1 when developer 2 is offering this feature. Developer 1 should just include it themselves.

ok

**Reply by Pravinhajare** [reply]
> > I'm currently working on a larger application and we divide it into broader components. To reflect these components we use a single-select, but several times issues arise which are related to multiple components at the same time.
> > A multi-select would better reflect this instead of insanely adding multiple single-selects or circumvent the absence of a multi-select by messing up issue labels which would have to be duplicated into multiple repositories.
> 
> kk

no

**Reply by Pravinhajare** [reply]
yes

**Reply by Pravinhajare** [reply]
> > > I'm currently working on a larger application and we divide it into broader components. To reflect these components we use a single-select, but several times issues arise which are related to multiple components at the same time.
> > > A multi-select would better reflect this instead of insanely adding multiple single-selects or circumvent the absence of a multi-select by messing up issue labels which would have to be duplicated into multiple repositories.
> > 
> > 
> > kk
> 
> no

sdf

**Reply by Pravinhajare** [reply]
> > > > I'm currently working on a larger application and we divide it into broader components. To reflect these components we use a single-select, but several times issues arise which are related to multiple components at the same time.
> > > > A multi-select would better reflect this instead of insanely adding multiple single-selects or circumvent the absence of a multi-select by messing up issue labels which would have to be duplicated into multiple repositories.
> > > 
> > > 
> > > kk
> > 
> > 
> > no
> 
> sdf

ghf

**Comment by amadeogallardo** [comment]
Multiselect would be great for us since we track which customer(s) requested/reported an issue.

**Comment by harsh04** [comment]
Multi-select is a much-needed field for managing projects right now. Similar to assignee

**Comment by ccolella-mdc** [comment]
Multi-select custom fields would be an invaluable feature!

**Comment by alexcombessie** [comment]
We need multi-select to track with customers asked which features. This is not possible without multi-select.
Unfortunately, we cannot use Github projects without this.

**Comment by umassahu** [comment]
Yes its much required , kindly implement the same so that we can run multiple test suites

**Comment by ntindle** [comment]
Would love this feature as well

**Comment by nicodard** [comment]
Requesting this as well. Critical to track which customers have requested for a feature. Should work like labels, but based on a different list.

**Reply by alexcombessie** [reply]
💯 I cannot believe that Projects was released in General Availability without this feature. Not being able to add multiple tags is a major blocker.

**Reply by Pravinhajare** [reply]
yes

**Comment by jaclynjessup** [comment]
Currently having to create a mashup of emojis when picking a type for the issues within our board. Multi-Select would significantly cut down on the amount of emoji mashup options I have to create
<img width="204" alt="Screen Shot 2022-10-07 at 1 01 31 PM" src="https://user-images.githubusercontent.com/23533895/194635433-c803510d-cab3-4626-a0e9-81f5db747442.png">

**Reply by av365app** [reply]
> Currently having to create a mashup of emojis when picking a type for the issues within our board. Multi-Select would significantly cut down on the amount of emoji mashup options I have to create <img alt="Screen Shot 2022-10-07 at 1 01 31 PM" width="204" src="https://user-images.githubusercontent.com/23533895/194635433-c803510d-cab3-4626-a0e9-81f5db747442.png">

That's sounds good.

**Reply by kiview** [reply]
That's a really cool workaround, thanks for sharing 👍

**Comment by kathy-phet** [comment]
We really need multi-select! And have been waiting almost a year. Can it make it on the project board?? Doesn't seem too hard. You already have the "Assignee" acting like a multi-select. Generallize that behavior and your set.

**Comment by kathy-phet** [comment]
This issue seems buried - but important ... it's one of your TOP requests for projects by up-votes, that hasn't been addressed.
https://github.com/community/community/discussions?discussions_q=label%3AIssues+sort%3Atop 

Can someone please add the appropriate tags "Projects", "Enhancement"? My team really needs this feature.

**Comment by ruanyl** [comment]
I understand that "issue" could have multi-labels, but this multi-select custom field is crucial for tasks that are not issue-ready. 

Perhaps there are performance/product concerns of supporting custom multi-select fields, but please do response to the community request as there are so many people voting and waiting for this feature 🙏 🙏 🙏

**Reply by d3rp3tt3** [reply]
I work with both private and public repos - these are pulled into one project. We can't use labels to track private stuff on public issues. It is really broken.

**Comment by kathy-phet** [comment]
@azenMatt - Tagging you in hopes that you might take a look at this request for a multi-select field on project boards. It is one of our top two requests for the project boards to be more useful - and it seems like the request is buried by newer requests - despite the large number of up-votes it has. Thanks for taking a look! And please add appropriate new labels.

**Comment by Jay-o-Way** [comment]
In contrast to all the "me too, me too" comments, it would be helpful to have specific use-case examples

**Reply by nicodard** [reply]
Plenty of use cases in the comments above.

**Comment by justyn** [comment]
There have been a few examples already in the earlier comments about the need for multi-select, but here are two of ours.

🤞🤞 for this to be discussed and added to the roadmap in some form.

### Identifying affected products

A product can consist of multiple repositories for various hardware and software elements, and some of these repos may be used by multiple products.

Currently there is no way in Projects to categorise which products are affected by a task/issue. Single-select is not suitable because an item can (and often does) affect multiple products.

### Identifying relevant teams

Similarly, an issue may (and again often does) affect multiple teams.

**Comment by justyn** [comment]
Like @kathy-phet I've noticed that this is now the [4th most upvoted discussion about the beta features that is still unanswered.](https://github.com/community/community/discussions?discussions_q=label%3AIssues+sort%3Atop+is%3Aunanswered+beta)

Surely they must have thought about it when they added single-select? I don't know how difficult multi-select would be to implement, but it seems to me that the design is done (single-select but, well, multi) and it's hard to imagine the backend changes being _that_ challenging in comparison to other features being discussed.

It's making me wonder if I'm misunderstanding something - is there perhaps a practical or ideological reason that multi-select should be avoided in this kind of interface? Or could similar functionality be being reserved as part of a higher-paying tier?

I'm absolutely not suggesting that this feature has to be a priority, of course Github can prioritise feature development as they wish. But it just seems a bit odd to me that there has been no response from them at all after over a year, given that so many people seem to have such an obvious need for it. I wonder if I'm missing something, or if there has been a response elsewhere?

**Reply by 2percentsilk** [reply]
Hey @justyn,

Thanks for taking the time to provide this feedback and also for your generous patience with us. Let me give an update here:
 
The short answer is adding new field types is not an easy lift for our team at the moment. In fact, it's actually quite hard as we learned from the last time we added a new field type. The team understands this feature would unlock a lot of value for our customers and it's high on the backlog but we need to land a few updates to our platform that would make adding new field types easier first. Trust that we are readily aware of this gap as Projects users ourselves and are regularly evaluating whether it makes sense to do this the "hard but sooner way." For the moment, we're making the choice to try to accelerate many new types of fields in the future by investing in our platform now. I know that doesn't solve the problem at hand for everyone needing multi-select but I hope this provides some context - we want to do this, it's just a matter of time and sequencing.

**Reply by justyn** [reply]
Thanks @2percentsilk for responding, and for the background context, it's appreciated!

It's very good to know that this feature is also seen as important within Github and that it forms part of your roadmap. I look forward to using it when it's ready (or even before!)

Thanks again

**Reply by d3rp3tt3** [reply]
Every other modern software project management system supports this capability for custom fields. I understand it's a lot of work, but please do prioritize it. We've considered moving away from projects just because of this one missing capability.

**Reply by MikeZiemak** [reply]
This is the major reason we're not using GH Projects at my company. Even when using it in my personal projects, sometimes it's difficult to attribute a particular issue into a single field, where multiple are needed.

Please prioritise it, if possible

**Reply by justyn** [reply]
Hi @2percentsilk is it possible to get a 9-month update on this? Any progress towards multi-select becoming available? Thank you!

**Reply by zsloane** [reply]
Hi @2percentsilk 

Of course investing in the platform to make adding new field types easier in the future was the right call. However, now that it is a full year later, is there any update on where this stands?

**Reply by riv-dalatorre** [reply]
happy birthday comment! Can we have an update please?

**Reply by dequejenn** [reply]
@2percentsilk - this update was 2 years ago. Any chance thos platform changes have landed and we are closer to having this feature now?

**Reply by beingminimal** [reply]
> Hey @justyn,
> 
> Thanks for taking the time to provide this feedback and also for your generous patience with us. Let me give an update here:
> 
> The short answer is adding new field types is not an easy lift for our team at the moment. In fact, it's actually quite hard as we learned from the last time we added a new field type. The team understands this feature would unlock a lot of value for our customers and it's high on the backlog but we need to land a few updates to our platform that would make adding new field types easier first. Trust that we are readily aware of this gap as Projects users ourselves and are regularly evaluating whether it makes sense to do this the "hard but sooner way." For the moment, we're making the choice to try to accelerate many new types of fields in the future by investing in our platform now. I know that doesn't solve the problem at hand for everyone needing multi-select but I hope this provides some context - we want to do this, it's just a matter of time and sequencing.

Is it that much harder than building llm from scratch, people building llm in 1 year, and such features in github are pending from last many years.
Can you please help to understand the difficulties? OR please share clarity like this is not going to be supported.
Awaiting your response. Thanks

**Reply by RecursiveSyntax** [reply]
Any further word or update? This is kind of a major blocker for required/basic expected functionality.

**Reply by SimoneZaza** [reply]
@evi-liu,

I saw your recent updates in other threads, any chance you have news on the multi-select feature? Would be super helpful to know if/when it’s coming!

Thanks for your work!

**Comment by VaZark** [comment]
I hope this can be prioritised sooner than later. While not it is not something ground-breaking, it is definitely a blocker. 

Almost all of our work is on GH, and given the advantages of a tight intégration, I'd like to move from my project organisation over here as well. Using something like clickup just for the sake of minor QoL features feels a bit wasteful

**Comment by VaZark** [comment]
Just realised that the `iteration` field is a single-select as well. This unfortunately renders the Gantt chart less useful for tasks that span multiple sprints/iterations :(

**Comment by hp8wvvvgnj6asjm7** [comment]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by wuebkers** [comment]
Would love this feature (or even a solution/workaround). In our case, we are building various reusable components that make up a website or an app, and want to be able to track all the pages that the component lives on and filter the "Tasks" view. A multi-select would be the easiest solution, but I am open to other workarounds.

**Reply by Julian-Dumitrascu** [reply]
Why [work around](https://www.microsoft.com/en-us/Investor/earnings/FY-2023-Q1/income-statements)?
We want developers to design software well enough, not to cause unnecessary costs.

**Comment by bytecode-praveen** [comment]
Multi-select Custom field is much required feature, I cannot believe that its not available with GitHub Project as it is a basic requirement for any project management tool

**Comment by vignesarul** [comment]
This is definitely a blocker. I dont have an option to mark and identify an issue that spans for multiple sprints.

**Comment by NicoFQ** [comment]
Estaría genial añadir esta funcionalidad pero nadie ha respondido desde hace más de dos años...

**Comment by BennyHoang** [comment]
This is much needed! 🙏

**Comment by mich-a-b** [comment]
In our project that we are developing, we need to indicate which modules are being affected by these bugs, and having a multi-select feature will be greatly beneficial when working with these use cases.

**Comment by heather-ashmore** [comment]
Also supporting the need for this multi-select field!

**Comment by ghost** [comment]
+1

**Comment by rubengalle** [comment]
Any updates to this feature request? It would be useful to be able to add multiple stakeholders to an issue so that they can be kept in the loop during or after development.

**Comment by GuillaumeFalourd** [comment]
+1 😄 ![1683133066365](https://github.com/community/community/assets/22433243/87d637dd-57aa-4f64-906b-de37d0a1cb5b)

**Reply by zsloane** [reply]
I guess we're not getting this basic functionality at this point. It's approaching 3 years with no meaningful update.

All focus clearly shifted to Copilot...

**Comment by icexuick** [comment]
Oh, man this really is needed it almost any type of application/flow.
Does anybody know if there is a official wish-list/forum where to go to regarding this?

**Comment by nicodard** [comment]
Can we get some sort of update here? Or is it easier to just switch to JIRA?

**Comment by vaz3r** [comment]
Multi-select is an integral part of the "projects" product and it must be prioritized, I still do not understand why it wasn't to date.

**Comment by jerradneff** [comment]
+1 on this.

**Comment by devison** [comment]
This is a feature we quickly realized we need as we are started working with GitHub Projects.  
Is it on a roadmap?

**Comment by dallenbaldwin** [comment]
+1 for my team. A couple of us are piloting GitHub Projects for an upcoming Milestone (the team uses ClickUp currently) and we quickly realized you can only add a multi-select like field once an issue has been created, and it has to be labels. This might be ok for some workflows, but we'd love a multi-select

**Comment by dgiacomo** [comment]
Can you get Github Copilot to implement this for us please?

**Comment by mikahpinegar** [comment]
My team desperately needs this feature. We are using GitHub projects for cross team planning and lots of issues get assigned to multiple teams. Only having a single select is a major frustration.

**Comment by NCatalani** [comment]
In hindsight this thread is ridiculous. Are there any more possible excuses from a technical standpoint for not having this 3 years later down the line?

**Comment by azenMatt** [comment]
I can assure you all this is still on the radar and I am painfully aware that this new field request (along with the adding of date fields like created / closed) is overdue. Right now we are blocked from working on this until we complete https://github.com/orgs/community/discussions/139936. Once the re-platforming there is finished we will pick these back up

**Comment by rodrigoslayertech** [comment]
YEAH!

![image](https://github.com/user-attachments/assets/328d2ed1-f37d-428a-bbcd-c7b047999d54)

I would like to see a new field for multiple choices.

To optimize space, the selected options could appear in an interleaved manner, that is, one by one, every x amount of time, in the same space, instead of showing one next to the other:

![MyVideo_2-ezgif com-crop](https://github.com/user-attachments/assets/3ad3608d-00f5-4122-9d45-8a5d8a69d8ac)

**Reply by zsloane** [reply]
Is this rolling out? How do we get access?

**Reply by rodrigoslayertech** [reply]
Not yet. It's just my suggestion. I edited the HTML to be able to display it.

**Reply by dequejenn** [reply]
You got me all excited that they finally did this!!!!

**Comment by luisarojas** [comment]
+1 !!!

**Comment by zbrown001** [comment]
+1 on this.

**Reply by LKA-Sendance** [reply]
+1

**Comment by agneszitte** [comment]
+1 for this issue, this feature is essential for effective project management and greatly benefits workflow. 
Please prioritize its implementation!

**Comment by dequejenn** [comment]
I saw back in [2023](https://github.com/orgs/community/discussions/50333#discussioncomment-5554948) this was in the backlog. Is there any update on when we might get this? I keep finding more and more use cases where this is needed. Thanks!

**Comment by riencoertjens** [comment]
🦗 chirp chirp...

**Comment by tu14145** [comment]
1+ for this, hope to see this soon! We want to use it to track multiple teams and PIs for an effort.

**Comment by chini007** [comment]
+1(00000) Please

**Comment by mari1647iv** [comment]
+1 from me on this

**Comment by Oscar-gg** [comment]
+1

**Comment by dequejenn** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by loky-lp** [comment]
Hey @azenMatt (hoping the ping doesn't bother you), but instead of writing the obnoxious "+1" I'd like to know, if possible, what is blocking this.
I've seen that [github.com/orgs/community/discussions/139936](https://github.com/orgs/community/discussions/139936) has been closed and it has been passed on the next release phase.
Once it will be fully released, is there anything that can block this? Can you give us a broad ETA?

**Reply by azenMatt** [reply]
Hi @loky-lp - you are correct https://github.com/orgs/community/discussions/139936 was our main blocker to being able to deliver this. 

Multi-select is still on our radar, however looking at the top requests in this repo: https://github.com/orgs/community/discussions/categories/projects-and-issues?discussions_q=is:open+category:%22Projects+and+Issues%22+sort:top+is:unanswered 

There are a couple of requests ahead of this that I would expect us to deliver first. The one most related to this one (in terms of new field types) is 'updated / created / closed dates'. I would expect us to work on that before delivering on multi select

**Reply by loky-lp** [reply]
Ty for the kind response

**Comment by snlwih** [comment]
+1

**Comment by ebattistini** [comment]
is this still a thing ? or GitHub is officially a Microsoft company?

**Reply by nsarang** [reply]
Yes

**Reply by ebattistini** [reply]
yes to what ? both ?

**Comment by dtitov** [comment]
What's the status of this? Are we getting it anytime soon?

**Comment by vb-dontenwill** [comment]
This would highly benefit our Versioning system.
Having only one target Version makes it so that Issues that target multiple versions have to do so implicitly.
The only workaround right now is to have multiple Fields "Version 1, Version 2, Version 3". This is, however, very redundant and inconsistent way to work.

**Comment by kimus** [comment]
If you have  custom fields, then it's obvious that multi select should be on of the types needed.
Please make it happen.

**Comment by RecursiveSyntax** [comment]
Any further word or update? This is kind of a major blocker for required/basic expected functionality. This should be an extremely high-priority for Project management, and would provide so much value, the lack of this feature is a huge gap ATM. 😞

**Reply by azenMatt** [reply]
I don't have a timeline update for you, but this is still on our radar as a top request. 

A higher priority right now is to close the gap with date fields, which would give you closed day, created date, and updated date in a project. Once we have that live (Q1 2026) we'll turn our focus to this one.

**Reply by linde-ibis** [reply]
Is this still the plan?

**Reply by aikenka** [reply]
+1 for the date fields 🍀

**Reply by jankoweb** [reply]
ad "closed day, created date, and updated date": yes, please! 

I'm about to write a workflow action for this, so should I wait? These fields already exist (as seen when trying to create a new column), but is there a roadmap for when they will be available in the Projects's GUI?

<img width="476" height="162" alt="image" src="https://github.com/user-attachments/assets/fa5ec4a6-7d70-48df-8978-86c6e93e1bd6" />

**Reply by linde-ibis** [reply]
@azenMatt have you got an update on both the multi-select field and date fields you said are planned for Q1?

**Comment by AlexVeeva** [comment]
We're moving over from Jira to GitHub Projects soon (hopefully) and this has been a major blocker for me too. Would have liked to have a `Team` and a `Components` custom field which could be multi-select but I've instead resorted to having the team as a label and the component as a single select (this will inevitably cause friction though)

Glad it sounds like it'll come in 2026 - but would also be nice to migrate single selects to multi too!

**Comment by amenity** [comment]
@azenMatt - any chance there's been progress since 10/25? 😬 🙏🏻

**Comment by triclambert** [comment]
Piling on...  this should be an obvious win with your customers.  Any chance you'll prioritize this any time soon?  🙏

**Comment by Arctiq-Ueb** [comment]
Obvious bump... Let's make it happen!

**Comment by JehlickaJo** [comment]
+1

**Comment by ezatti** [comment]
+1

**Comment by sebaaaz** [comment]
Any updates on this? Thanks!

**Comment by JulianNymark** [comment]
multi-select would be nice :) please add it!

**Comment by amenity** [comment]
@labudis @broccolinisoup - I realize that this highly sought-after request applies to custom project fields, not issue fields. Should I open a new thread to get it in the issue fields backlog?

**Reply by labudis** [reply]
We are planning to introduce this feature across both Issue fields and Project fields at the same time - work is underway!

**Comment by GregJohnStewart** [comment]
So I now see the multi select field as an option, but there is no ui for actually adding options? Nothing changes when I select MultiSelect (like with single select). Thoughts? Is it just me?

**Reply by aikenka** [reply]
I don't even see multi select as an option - still just single select. Unless it's showing up somewhere else?

<img width="230" height="390" alt="image" src="https://github.com/user-attachments/assets/43b80c18-47c0-44cb-a3a7-cb99672fda52" />

Have you found a window into the near future 🔮😊 really hoping this is coming soon!!

**Reply by beingminimal** [reply]
its in issue fields and not in project fields. thx

**Reply by amenity** [reply]
Looks like it's not working yet, but hopefully this means it's coming soon!
<img width="2170" height="1284" alt="Screenshot 2026-06-11 at 10 00 56 AM" src="https://github.com/user-attachments/assets/ecc49776-bdfb-4f39-ba8a-9c1e8e619fa4" />

(There's no way to input values once you've selected multi-select. I tried starting it in single-select and then switching, but that didn't help.)

**Reply by beingminimal** [reply]
Yes, right its not working. Lets wait. Thanks

**Reply by labudis** [reply]
Hey folks, we are actively working on the multi-select feature to bring it to both Issue and Project fields. This control accidentally "escaped" before it was ready and I am very sorry for the confusion. It's not long now!

**Reply by amenity** [reply]
I hate to ask the dreaded question, but can you estimate a release date for this, @labudis? Basically just wondering whether we're looking at weeks or months. 🙈

**Reply by amenity** [reply]
@aconsuegra & @broccolinisoup - any word on multi-select?

**Reply by aconsuegra** [reply]
@amenity We are testing it internally now, and hope to have it ready in production in a couple of weeks

**Reply by amenity** [reply]
Excellent! Thanks for the quick reply, @aconsuegra 

& Lmk if you need a pre-release tester. I can write some solid bug reports. ;)

**Comment by Lindbergo** [comment]
Hey! Does anyone know where this is upto? This would really improve the use of projects for us :)

**Comment by sahanakothandapani1** [comment]
Multi-select fields would also be helpful for educational and student projects. While exploring open source, I've come across issues that fit into multiple categories at the same time, such as documentation, frontend, and testing. Having a multi-select field would make organizing these issues much simpler than creating multiple separate fields or relying solely on labels.

**Comment by sunnyi101** [comment, marked as answer]
This is live now - https://github.blog/changelog/2026-07-23-multi-select-fields-for-projects-and-issues-in-public-preview/!

**Reply by aikenka** [reply]
Thanks @sunnyi101 so glad to see this implemented, starting to use it now!

As I work to update items to use my new multi-select field, is there a way in project table view to copy selected values from one row to others, as we can do with single-select values? I'm pressing ctrl-c on a cell, but it doesn't seem to copy the selected values.

**Reply by aikenka** [reply]
I tried setting the multi-select field value via Claude.ai in VSCode (I have project analysis and cleanup scripts running there), but we hit some snags. Details per Claude - I'll open a bug report as well with full details and reference IDs.

Summary
A field created in the web UI as a multi-select ("Product (multi-select)") cannot be
served by the public GraphQL API. Any query that must serialize the field's
DEFINITION, or an ITEM VALUE of it, returns a top-level 500-style error:
  {"errors":[{"message":"Something went wrong while executing your query ..."}]}
(null data, non-zero exit) instead of a clean schema/validation error. Introspection
shows no ProjectV2 multi-select types and no multi-select variant on the
ProjectV2FieldValue mutation input, so the type appears unsupported in the public
schema — but a query merely touching it should not 500.

Expected
The field/value serializes, OR a clean documented GraphQL error. A single
unsupported field should not 500 an entire query or page.

Actual
Top-level "Something went wrong" (null data). Field id/option ids are therefore
unreadable via the API, and updateProjectV2ItemFieldValue has no multi-select input,
so the field can't be read, written, or cleared through GraphQL — and any automation
sweeping ProjectV2.items fieldValues crashes when it hits a populated item.

Ask
1. (Bug) Return a graceful error instead of 500 when a query touches this field type.
2. (Feature) Expose ProjectV2 multi-select fields (read + write) in the public
   GraphQL API.

**Reply by vb-dontenwill** [reply]
Thanks for implementing.

I found a bug:

> When deselecting the last choice, the Multiselect turns empty for a short time, but returns the previous value shortly after, rendering it unable to be emptied after a value has been set. If 3 values were selected before closing, all 3 are returned.

1.

<img width="268" height="284" alt="image" src="https://github.com/user-attachments/assets/9f9973d8-a6b5-4376-b0f8-a63b880827e0" />

2.

<img width="276" height="278" alt="image" src="https://github.com/user-attachments/assets/5106f873-a7ee-4f52-ac17-90607ee50411" />

3.

<img width="227" height="52" alt="image" src="https://github.com/user-attachments/assets/eaf0c573-671f-473a-83ca-a506bbd3d37e" />

**Reply by ayy-bc** [reply]
@aikenka we just shipped GQL support for multi select so that should be live now :) 

And copy pasting is ~currently being~ fixed ~so we should have support for that shortly (by end of day)~ as well

**Reply by aikenka** [reply]
Excellent on both counts - thanks @ayy-bc! I've tried the copy/paste successfully, and Claude is happily GraphQL'ing away at updating all the other issues on my board.

Will we be able to "group by" these fields as well?

**Reply by ayy-bc** [reply]
@vb-dontenwill this (not able to clear options from issue page) should be fixed now 😄 thank you so much for being early users and reporting the bugs 🙏

**Reply by aikenka** [reply]
Will we be able to "Slice by" these fields as well?

**Reply by amenity** [reply]
@ayy-bc - +1 to @aikenka's question - I love "Slice by" and "Group by" for the GHP table view. I realize it's a bit illogical to have this work for a multi-select field because things will show up twice, so the number of rows will not match the number of distinct issues... but I think I'd still like the option to navigate issues in that way.

**Reply by ayy-bc** [reply]
@aikenka @amenity thank you so much for being early users and for the feature requests. We've shipped support for slicing and grouping for these fields. 

For grouping however, its only supported in table view and we disabled drag and drop when grouped by multi select values. That is because drag and drop semantics can get confusing when items appear in multiple groups. You can still add items in the group and it will assign the group value to the newly added item (and this works for slice by as well as filters).

**Reply by aikenka** [reply]
Thanks @ayy-bc it's working great!

We mostly work in table views that are auto-sorted (no drag-and-drop, we sort by priority/status), so all good for us. And yep the values are being added to new items as expected.

"Group by" seems to work in gantt and kanban (swimlanes) as well 🥳 

One weirdness with slicing - the slicing works fine, but any views I've modified to use the new field for slicing don't seem to save - I click to save view, all seems fine, but the blue-dot-need-to-save-hint never clears.

I tried "save changes to new view" - the new view didn't include the slicer at all, and the original view lost it too.

<img width="313" height="427" alt="image" src="https://github.com/user-attachments/assets/3ceb2391-d4da-4a5f-bd26-78910db3d604" />

**Reply by ayy-bc** [reply]
@aikenka should be fixed now 😃

**Reply by aikenka** [reply]
Yep, working now, thanks!

**Reply by amenity** [reply]
Yep, love it! Thanks, @ayy-bc

**Reply by wafer-bw** [reply]
We can't sort by multi-select fields? They have an index like single-select so we should be able to sort by it but its greyed-out. Also looks like selecting a multi-select field on a board view doesn't render them on issue cards.

**Comment by amenity** [comment]
Working beautifully so far. Thanks!

**Comment by grendes-wunder** [comment]
I try to use such a Field on Organization level, as Issue field.
I have configured a Project Boards' view to display this new field ("Component") on the Issue's card but no luck.
Most other fields *do* show up. What do I miss? Could this be a bug / not supported on Issue cards?

<img width="351" height="893" alt="image" src="https://github.com/user-attachments/assets/461a7ece-0801-466a-aab9-bcd19d1c73ea" />

<img width="371" height="92" alt="image" src="https://github.com/user-attachments/assets/e1c5ea47-c893-4dc6-b1b0-1ae06094199e" />

**Comment by Arctiq-Ueb** [comment]
For anyone using GraphQL to call multi-select fields, you should refer to 'ProjectV2MultiSelectField'.

<details><summary>example</summary>

```
gh api graphql -f query='
{
  node(id: "#{...}") {
    ... on ProjectV2 {
      fields(first: 50) {
        nodes {
          ... on ProjectV2MultiSelectField {
            id
            name
            multiSelectOptions {
              id
              name
            }
          }
        }
      }
    }
  }
}
'
```

</details>
